src/dune_rules/modules.ml

```diff
@@ -1131,3 +1166,14 @@ let source_dirs =
       Module.sources m
       |> List.fold_left ~init:acc ~f:(fun acc f ->
              Path.Set.add acc (Path.parent_exn f)))
+
+let canonical_path t (group : Group.t) m =
+  let path =
+    let path = Module.path m in
+    match Module_name.Map.find group.modules (Module.name m) with
+    | None | Some (Group.Module _) -> path
+    | Some (Group _) -> path |> List.rev |> List.tl |> List.rev
```

Comment:
what's this case here? why are we removing the last path segment?

Reply:
It's for interface modules. A module like `foo/foo.ml` would have a path like `[ "Foo"; "Foo" ]` so the last element has to be removed to get the canonical path.